### ISSUE TITLE
Better UI for SSH key management

### ISSUE THREAD

**Issue by maks** [maks/MGit#368]
As illustrated by #326 the UX of the current SSH key management UI is dreadful. Need better UI for showing public/private key contents and also need to use proper list UI instead of trying to reuse the file browser activity.

**Comment by martinellison** [comment]
Agree, I am trying to navigate to an existing key file, and all I am getting is "cannot open this file" errors.

**Comment by spaetz** [comment]
Even better would be integration with something like OpenKeyChain. It caches passwords securely and with configurable timeouts, and it does not require my private SSH key to be lying unprotected somewhere on the Android file system.

**Comment by kaeptmblaubaer1000** [comment]
No, OpenKeychain stores PGP, not SSH keys. Also, that would probably either be complicated to implement or put a dependency on OpenKeychain, or even both.

**Comment by spaetz** [comment]
DOH, ignore my comment, brain wasn't working yet...

**Comment by nicobo** [comment]
@kb-1000 yes, OpenKeychain can use an authentication subkey if a PGP key as an SSH key.

You can look at [passwordstore's Android client](https://github.com/android-password-store/Android-Password-Store#readme) : it allows to chose OpenKeychain as the SSH Key provider.

Once you know that you must add an auth. subkey, it's a very simple and working UI.

**Comment by maisiliymBridj** [comment]
As @nicobo pointed out, only to add that [okcAgent](https://github.com/DDoSolitary/OkcAgent) can forward openkeychains ssh auth to termux, which may be useful for mgit.

**Comment by maks** [comment]
@nicobo thanks for the pointer! 

I hadn't come across OpenKeyChain app before.

Doing a bit of research, looks like the relevant bit of work is in [this commit of Android-Password-Store](https://github.com/android-password-store/Android-Password-Store/pull/486) and specifically [the SshApiSessionFactory](https://github.com/android-password-store/Android-Password-Store/blob/25b426157417c97c61c3d2cc0d9fa556f337f22d/app/src/main/java/com/zeapo/pwdstore/git/config/SshApiSessionFactory.java) is at the heart of getting the integration between JSCH which is the ssh lib that JGit (and hence MGit) uses and OpenKeyChain's API.

This would definitely be an ideal way for MGit not to have anything to do with managing users ssh keys, though I must say it does look quite convoluted per the above implementation, but the good thing is that at least they already have an implementation done and working with a compatible license. I'd be happy if anyone wants to have a go at implementing this for MGit.
